### ISSUE TITLE
autogen prints out warnings

### ISSUE THREAD

**Issue by dongahn** [flux-framework/flux-core#1869]
```
quartz770{dahn}79: autogen.sh
Running libtoolize --automake --copy ...
Running autoreconf --verbose --install
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal -I config
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy
autoreconf: running: /usr/bin/autoconf
autoreconf: running: /usr/bin/autoheader
autoreconf: running: automake --add-missing --copy --no-force
configure.ac:48: installing 'config/compile'
configure.ac:9: installing 'config/config.guess'
configure.ac:9: installing 'config/config.sub'
configure.ac:22: installing 'config/install-sh'
configure.ac:22: installing 'config/missing'
doc/man3/Makefile.am: installing 'config/depcomp'
parallel-tests: installing 'config/test-driver'
src/bindings/python/_flux/Makefile.am:39: installing 'config/py-compile'
src/bindings/python/_flux/Makefile.am:54: warning: variable 'nodist__job_la_SOURCES' is defined but no program or
src/bindings/python/_flux/Makefile.am:54: library has '_job_la' as canonical name (possible typo)
src/bindings/python/_flux/Makefile.am:9: warning: variable 'AM_LIBADD' is defined but no program or
src/bindings/python/_flux/Makefile.am:9: library has 'AM' as canonical name (possible typo)
src/bindings/python/_flux/Makefile.am:56: warning: variable '_job_la_LIBADD' is defined but no program or
src/bindings/python/_flux/Makefile.am:56: library has '_job_la' as canonical name (possible typo)
src/bindings/python/_flux/Makefile.am:57: warning: variable '_job_la_DEPENDENCIES' is defined but no program or
src/bindings/python/_flux/Makefile.am:57: library has '_job_la' as canonical name (possible typo)
```

Are the warning false positives?

**Comment by chu11** [comment]
They appear to be harmless, but I'm not sure.  Digging into the commit history:

The ```AM_LIBADD``` was added by @trws in 9bccd1fdc749ddff31d348271737bd984457ac9d.  The comment in that commit is "clean up debug prints and annotate substitutions".  Possible left over from some other experimenting?

The ```_job_la``` ones appear to be that in 451adf765455e663d9d0b8bb9159ef7a31dbd092, the ```_job.la``` was removed from ```fluxso_LTLIBRARIES```.   Or ```nodist__job_la_SOURCES``` (and similar ```_job.la``` vars) were supposed to be removed but weren't?

**Comment by trws** [comment]
That was moving the python_ldflgs from the _ldflags var to libadd to attempt to fix the lib tool delink issue on Sierra.


---
Sent from Workspace ONE Boxer<https://whatisworkspaceone.com/boxer>

On December 18, 2018 at 5:42:13 PM PST, Al Chu <notifications@github.com> wrote:

They appear to be harmless, but I'm not sure. Digging into the commit history:

The AM_LIBADD was added by @trws<https://github.com/trws> in 9bccd1f<https://github.com/flux-framework/flux-core/commit/9bccd1fdc749ddff31d348271737bd984457ac9d>. The comment in that commit is "clean up debug prints and annotate substitutions". Possible left over from some other experimenting?

The _job_la ones appear to be that in 451adf7<https://github.com/flux-framework/flux-core/commit/451adf765455e663d9d0b8bb9159ef7a31dbd092>, the _job.la was removed from fluxso_LTLIBRARIES. Or nodist__job_la_SOURCES (and similar _job.la vars) were supposed to be removed but weren't?

**Comment by SteVwonder** [comment]
I'll try and poke at this on the plane ride tomorrow.

**Comment by SteVwonder** [comment]
After looking at it, I believe the `_job` references are vestiges from before #1795, when every major chunk of flux-core got it's own cffi-generated shared library.  Now that everything ends up in a single shared library (i.e., `_core.so`), I think these can be safely removed.  Will double check on my flight, and post a PR after landing.

**Comment by trws** [comment]
That's right, I thought I had gotten all of that, but must have made a mistake. The other with the libadd is probably that it needs to be target prefixed rather than AM_, no clue why I didn't get warnings about it though. Maybe testing with an old version of the auto tools or something?

On 20 Dec 2018, at 9:17, Stephen Herbein wrote:

After looking at it, I believe the _job references are vestiges from before #1795<https://github.com/flux-framework/flux-core/pull/1795>, when every major chunk of flux-core got it's own cffi-generated shared library. Now that everything ends up in a single shared library (i.e., _core.so), I think these can be safely removed. Will double check on my flight, and post a PR after landing.
